### PR TITLE
feat: add repeats and custom id sequences

### PR DESCRIPTION
**Pull request by sapiderman**
### What this PR does
Implements #147 to add run options
1. Added -r  --repeat flag
2. Added -i  --id flag   
3. Add documentations in guides

### Implementation 
1. Added an id feeder to handle user specified probe.id sequences
2. Added counter checking for repeat control

## How to test
1. Running `monika` or `bin/run` should work normally
2. `monika -r x` where x is any number, should repeat just x times
3. `monika -i 1,2,3` should run probe id 1, 2 and then 3. Sequence can be any valid id sequence (ex: 1,2,3 or 3,2,1 or 1,2,1) 
4. `monika -r 3 -i 1,2` should run probe id 1 and 2 repeating 3 times
5. any invalid probe.id should be caught

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -60,14 +60,21 @@ function getDefaultConfig() {
 class Monika extends Command {
   static description = 'Monika command line monitoring tool'
 
+  static examples = [
+    'monika',
+    'monika --1ogs',
```
Comment (armedi) [inline src/index.ts:65]:
```suggestion
    'monika --logs',
```

File: src/index.ts [hunk 2]
```diff
@@ -146,9 +165,7 @@ class Monika extends Command {
         const startupMessage = this.buildStartupMessage(config, flags.verbose)
 
         this.log(startupMessage)
-
-        // Loop through all probes
-        abortCurrentLooper = loopProbes(config)
+        idFeeder(config, Number(flags.repeat), flags.id)
```
Comment (armedi) [inline src/index.ts:168]: when abortCurrentLooper is removed, monika will not be able to update it's configuration if new configuration is received from Symon

**Review by armedi — CHANGES_REQUESTED** [review]

File: src/looper.ts [hunk 3]
```diff
@@ -95,18 +95,17 @@ export function isIDValid(config: Config, ids: string): boolean {
   return true
 }
 
+let isAborted = false
```
Comment (armedi) [inline src/looper.ts:98]: moving this variable to global scope will cause the value of isAborted to stay `true` after it's first set to true. 

This means that when monika receive new configuration from Symon, it clear probe interval from the previous configuration and the new interval for the new configuration will also get aborted immediately. Or in other word, the next configurations will never run.

**Review by armedi — CHANGES_REQUESTED** [review]

**Comment by dennypradipta** [comment]
It seems like that some iteration did not run the probe according to the order:

1. Running with `./run -r 3`
image

In the first iteration it is being run using the correct order: 1,2,3. In the second iteration, it is being run in wrong order: 3,1,2

2. Running with `./run -3 -i 1,2`
image

In the first iteration it is being run using the wrong order: 2,1. In the second iteration, it is being run in correct order: 1,2

File: src/looper.ts [hunk 4]
```diff
@@ -69,33 +70,120 @@ function sanitizeProbe(probe: Probe, index: number): Probe {
...
+  }
+  isAborted = false
 
   const abort = () => {
     isAborted = true
   }
 
+  // doing custom sequences?
+  if (ids) {
+    const idSplit = ids.split(',')
```
Comment (nicnocquee) [inline src/looper.ts:163]: need to trim the elements

**Review by nicnocquee — APPROVED** [review]

**Review by armedi — APPROVED** [review]

**Pull request merged**